lib/server.js

```diff
@@ -1,9 +1,76 @@
 'use strict';
 
-var sync = require('fh-wfm-sync/lib/server')
-  , config = require('./config');
+var sync = require('fh-wfm-sync/lib/server'),
+  config = require('./config'),
+  shortid = require('shortid');
 
 
+/**
+ * Function to generate topic used by the mediator to request, subscribe and publish to.
+ *
+ * @param {string} topicPrefix Defines the topic name.
+ * @param {string} action Defines which CRUDL action is to be used.
+ * @param {string} prefix Prefix used to add to the start of the topic to determine if request was done.
+ * @param {string} uid Unique identifier to identify unique topics.
+ * @returns {string} topic Generated topic to be used by the mediator
+ */
+function generateTopic(topicPrefix, action, prefix, uid) {
+  var topic = "";
+
+  if (prefix) {
+    topic = topic + prefix + ':';
+  }
+  topic = topic + topicPrefix + config.datasetId + ':' + action;
+  if (uid) {
+    topic = topic + ':' + uid;
+  }
+
+  return topic;
+}
```

Comment:
I mention deduping dependencies because this repo doesn't have the mediator as a dependency yet, just kind of a peer depenendy, but deduping would mean only a single copy would be installed on disk anyway, so no extra filesize.